src/components/vive-controls.js

```diff
@@ -43,36 +45,129 @@ module.exports.Component = registerComponent('vive-controls', {
     this.onButtonDown = function (evt) { self.onButtonEvent(evt.detail.id, 'down'); };
     this.onButtonUp = function (evt) { self.onButtonEvent(evt.detail.id, 'up'); };
     this.onModelLoaded = bind(this.onModelLoaded, this);
+    this.controllerPresent = false;
+    this.everGotGamepadEvent = false;
+    this.lastControllerCheck = 0;
   },
 
-  play: function () {
+  startListening: function () {
     var el = this.el;
+    // console.log('startListening');
     el.addEventListener('buttonchanged', this.onButtonChanged);
     el.addEventListener('buttondown', this.onButtonDown);
     el.addEventListener('buttonup', this.onButtonUp);
     el.addEventListener('model-loaded', this.onModelLoaded);
   },
 
-  pause: function () {
+  stopListening: function () {
     var el = this.el;
+    // console.log('stopListening');
     el.removeEventListener('buttonchanged', this.onButtonChanged);
     el.removeEventListener('buttondown', this.onButtonDown);
     el.removeEventListener('buttonup', this.onButtonUp);
     el.removeEventListener('model-loaded', this.onModelLoaded);
   },
 
-  update: function () {
+  checkIfControllerPresent: function () {
```

Comment:
(same comment as above)

if you look carefully, you'll see that `checkIfControllerPresent` is not common code to be readily moved to a common component; it necessarily does something different (i.e. vendor specific) for `oculus-touch-controls` and `vive-controls` even though both of them enumerate gamepads.  

If OpenVR could be trusted to correctly return hand attribute, then it might be possible to combine logic, but at the moment that can cause Vive owners to be provided two right hands and zero left hands at times, clearly not ideal.  

Ideally all these would use the same controller list for example, but I don't see how that is doable with current code structures.  Will give it some thought.